etw/sample/sample.go

```diff
@@ -0,0 +1,53 @@
+// Shows a sample usage of the ETW logging package.
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+
+	"github.com/Microsoft/go-winio/etw"
+	"github.com/sirupsen/logrus"
+
+	"golang.org/x/sys/windows"
+)
+
+func callback(sourceID *windows.GUID, isEnabled uint32, level byte, matchAnyKeyword uint64, matchAllKeyword uint64, filterData uintptr) {
+	fmt.Printf("Callback: isEnabled=%d, level=%d, matchAnyKeyword=%d\n", isEnabled, level, matchAnyKeyword)
+}
+
+func main() {
+	providerID := windows.GUID{0xdd2062c6, 0x5d1b, 0x4a0f, [8]uint8{0xbd, 0xb9, 0x22, 0x28, 0xbc, 0xb1, 0x07, 0x7c}}
```

Comment:
What GUID is this? Can you give a comment here?

Reply:
Every provider needs to define its own GUID which uniquely identifies it to ETW. I'll add a comment explaining this.

Also, tracelogging has an algorithm to generate the provider GUID based on the provider name. Once I get that implemented, we can just pass a name in most cases, and not worry about the GUID here.

Reply:
See my other comment about removing GUID. We should do that instead of requiring people to pass a `GUID` here.

Reply:
I'm not 100% sure on this. I remember TraceLoggingProvider.h making it easy to avoid passing a GUID, and then later that was removed/heavily discouraged. Before we automatically generate a GUID based on the name, we should understand why that was so.

Reply:
Got a response:

> I don’t know of any reason not to rely on auto-generation. It’s also publicly documented (as EtwGuid) and used in EventSource as the default GUID generator for providers.

Reply:
OK good deal.